sys/kern/src/descs.rs

```diff
@@ -0,0 +1,149 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+//! Descriptor types, used to statically define application resources.
+
+use crate::umem::USlice;
+
+pub(crate) const REGIONS_PER_TASK: usize = 8;
+
+/// Indicates priority of a task.
+///
+/// Priorities are small numbers starting from zero. Numerically lower
+/// priorities are more important, so Priority 0 is the most likely to be
+/// scheduled, followed by 1, and so forth. (This keeps our logic simpler given
+/// that the number of priorities can be reconfigured.)
+///
+/// Note that this type *deliberately* does not implement `PartialOrd`/`Ord`, to
+/// keep us from confusing ourselves on whether `>` means numerically greater /
+/// less important, or more important / numerically smaller.
+#[derive(Copy, Clone, Debug, Eq, PartialEq, Default)]
+#[repr(transparent)]
+pub struct Priority(pub u8);
+
+impl Priority {
+    /// Checks if `self` is strictly more important than `other`.
+    ///
+    /// This is easier to read than comparing the numeric values of the
+    /// priorities, since lower numbers are more important.
+    pub fn is_more_important_than(self, other: Self) -> bool {
+        self.0 < other.0
+    }
+}
+
+/// Record describing a single task.
+#[derive(Clone, Debug)]
+pub struct TaskDesc {
+    /// Identifies memory regions this task has access to, by index in the
+    /// `RegionDesc` table. If the task needs fewer than `REGIONS_PER_TASK`
+    /// regions, it should use remaining entries to name a region that confers
+    /// no access; by convention, this region is usually entry 0 in the table.
```

Comment:
This comment seems out of date, because the data is now here!

Reply:
Hmmm, will rewrite, thanks

Reply:
Fixed.

Reply:
We talked about this in chat, but, that is in fact the fixed comment -- references to indices removed.